flink-table-store-core/src/main/java/org/apache/flink/table/store/table/AppendOnlyFileStoreTable.java

```diff
@@ -62,7 +60,7 @@ public TableScan newScan(boolean incremental) {
         return new TableScan(scan, schema, store.pathFactory()) {
             @Override
             protected void withNonPartitionFilter(Predicate predicate) {
-                scan.withValueFilter(predicate);
+                ((AppendOnlyFileStoreScan) scan).withFilter(predicate);
```

Comment:
Any update?

Reply:
We can't remove this casting even if `AppendOnlyFileStore` returns `AppendOnlyFileStoreScan`. Because in `TableScan` class, `scan` is marked as a `FileTableScan` type. We have to use casting to change it into a subclass.

Reply:
```
@Override
    public TableScan newScan() {
        AppendOnlyFileStoreScan scan = store.newScan();
        return new TableScan(scan, schema, store.pathFactory()) {
            @Override
            protected void withNonPartitionFilter(Predicate predicate) {
                scan.withFilter(predicate);
            }
        };
    }
```